### DISCUSSION TITLE
I added korean language code. How to train base model?

### DISCUSSION THREAD

**Discussion by jwj7140** [fishaudio/Bert-VITS2#288] [Q&A]
Hello. First, thank you for amazing project.
I added some code about supporting korean in this repository: [https://github.com/jwj7140/Bert-VITS2-Korean](https://github.com/jwj7140/Bert-VITS2-Korean).
After that, I've tryed to train model with `train_ms.py`. But it did not work well.
```
Using noise scaled MAS for VITS2
Using duration discriminator for VITS2
INFO:models:Loaded checkpoint 'Data/kss/models/DUR_0.pth' (iteration 0)
ERROR:models:enc_p.emb.weight is not in the checkpoint
ERROR:models:enc_p.tone_emb.weight is not in the checkpoint
ERROR:models:enc_p.language_emb.weight is not in the checkpoint
ERROR:models:enc_p.ko_bert_proj.weight is not in the checkpoint
ERROR:models:enc_p.ko_bert_proj.bias is not in the checkpoint
ERROR:models:emb_g.weight is not in the checkpoint
INFO:models:Loaded checkpoint 'Data/kss/models/G_0.pth' (iteration 0)
INFO:models:Loaded checkpoint 'Data/kss/models/D_0.pth' (iteration 0)
```
I think existing base model(like [here](https://huggingface.co/OedoSoldier/Bert-VITS2-2.3)) is useless in this case. So...I have to try to train new base model in korean.
Can you introduce the way of training base model in detailed?

**Comment by thundax-lyp** [comment]
this issue occurs when loading the model. see load_checkpoint() in utils.py. 
```
assert saved_state_dict[k].shape == v.shape, (
    saved_state_dict[k].shape,
    v.shape,
)
```
maybe you are loading a wrong base model. you had better train the base model by yourself while add a new language

**Reply by jwj7140** [reply]
Yeah. I think so too. But despite many attempts to train new base model(with train_ms.py), it can't generate proper voice until now.
Here is output .wav file of my new base model in english: https://drive.google.com/file/d/1Pn0f3tuJJGUW_gSzVCg0QRZhks1o2mpt/view?usp=sharing

Are there some special ways to train base model? If not, how many epochs are required to train base model?
